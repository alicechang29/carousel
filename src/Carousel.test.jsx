import { describe, it, expect } from "vitest";

import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";

describe("carousel works", function () {
  it("renders without crashing", function () {
    render(<Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />);
  });

  it("works when you click on the right arrow", function () {
    const { container } = render(
      <Carousel
        photos={TEST_IMAGES}
        title="images for testing"
      />
    );
    // expect the first image to show, but not the second
    expect(
      container.querySelector('img[alt="testing image 1"]')
    ).toBeInTheDocument();
    expect(
      container.querySelector('img[alt="testing image 2"]')
    ).not.toBeInTheDocument();

    // move forward in the carousel
    const rightArrow = container.querySelector(".bi-arrow-right-circle");
    fireEvent.click(rightArrow);

    // expect the second image to show, but not the first
    expect(
      container.querySelector('img[alt="testing image 1"]')
    ).not.toBeInTheDocument();
    expect(
      container.querySelector('img[alt="testing image 2"]')
    ).toBeInTheDocument();
  });

  it("works when you click on the left arrow", function () {
    const { container, debug } = render(
      <Carousel
        photos={TEST_IMAGES}
        title="images for testing"
      />
    );
    debug(container);
    // expect the second image to show, but not the first

    //move forward in the carousel
    const rightArrow = container.querySelector(".bi-arrow-right-circle");
    fireEvent.click(rightArrow);

    expect(
      container.querySelector('img[alt="testing image 2"]')
    ).toBeInTheDocument();
    expect(
      container.querySelector('img[alt="testing image 1"]')
    ).not.toBeInTheDocument();

    // move backward in the carousel
    const leftArrow = container.querySelector(".bi-arrow-left-circle");
    fireEvent.click(leftArrow);

    // expect the first image to show, but not the second
    expect(
      container.querySelector('img[alt="testing image 2"]')
    ).not.toBeInTheDocument();
    expect(
      container.querySelector('img[alt="testing image 1"]')
    ).toBeInTheDocument();
  });

  it("hides left arrow when on the first image", function () {
    const { container } = render(
      <Carousel
        photos={TEST_IMAGES}
        title="images for testing"
      />
    );
    // expect the right arrow to show but not the left arrow
    expect(
      container.querySelector(".bi-arrow-right-circle")
    ).toBeInTheDocument();
    expect(
      container.querySelector(".bi-arrow-left-circle")
    ).not.toBeInTheDocument();

  });

  it("hides right arrow when on the last image", function () {
    const { container } = render(
      <Carousel
        photos={TEST_IMAGES}
        title="images for testing"
      />
    );
    // expect the left arrow to show but not the right arrow
    expect(
      container.querySelector(".bi-arrow-right-circle")
    ).toBeInTheDocument();
    expect(
      container.querySelector(".bi-arrow-left-circle")
    ).not.toBeInTheDocument();
  });


});


describe("snapshot tests for stability", function () {

  it("matches initial carousel on start", function () {
    const { container } = render(<Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />);
    expect(container).toMatchSnapshot();
  });
  // end

  it("matches hiding right arrow when on last image", function () {
    const { container } = render(<Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />);
    const rightArrow = container.querySelector(".bi-arrow-right-circle");

    fireEvent.click(rightArrow);
    fireEvent.click(rightArrow);

    expect(container).toMatchSnapshot();
  });
  // end

  it("matches having both arrows when not on first or last image", function () {
    const { container } = render(<Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />);
    const rightArrow = container.querySelector(".bi-arrow-right-circle");
    fireEvent.click(rightArrow);

    expect(container).toMatchSnapshot();
  });
});