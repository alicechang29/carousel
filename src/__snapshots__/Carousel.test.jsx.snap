// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot tests for stability > matches having both arrows when not on first or last image 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`snapshot tests for stability > matches hiding right arrow when on last image 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 3
        </h4>
        <img
          alt="testing image 3"
          class="Card-image"
          src="test3.com"
        />
        <small
          class="Card-small"
        >
          Image 
          3
           of 
          3
          .
        </small>
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot tests for stability > matches initial carousel on start 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 1
        </h4>
        <img
          alt="testing image 1"
          class="Card-image"
          src="test1.com"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;
